fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -392,20 +395,24 @@
     @OneToOne(cascade = CascadeType.ALL, mappedBy = "loan", optional = true, orphanRemoval = true, fetch=FetchType.EAGER)
     private LoanTopupDetails loanTopupDetails;
 
+    @ManyToMany(fetch = FetchType.EAGER)
```

Comment:
I think it is OneToMany relationship !!
it possible to change fetchtype to lazy and achieve the expected results?

Reply:
About this change I think that ManyToMany is okay as we use a third table to join the loan and rate tables just like in the LoanProduct class. I'll make the change and test if the same result can be achieved.